src/mono/wasm/runtime/library_mono.js

```diff
@@ -536,109 +540,424 @@ var MonoSupportLib = {
-					var memory = Module._malloc(asm.length);
-					var heapBytes = new Uint8Array(Module.HEAPU8.buffer, memory, asm.length);
-					heapBytes.set (asm);
-					mono_wasm_add_assembly (file_name, memory, asm.length);
-
-					//console.log ("MONO_WASM: Loaded: " + file_name);
-					--pending;
-					if (pending == 0) {
-						MONO.loaded_files = loaded_files;
-						var load_runtime = Module.cwrap ('mono_wasm_load_runtime', null, ['string', 'number']);
-
-						console.log ("MONO_WASM: Initializing mono runtime");
-						if (ENVIRONMENT_IS_SHELL || ENVIRONMENT_IS_NODE) {
-							try {
-								load_runtime (vfs_prefix, enable_debugging);
-							} catch (ex) {
-								print ("MONO_WASM: load_runtime () failed: " + ex);
-								var err = new Error();
-								print ("MONO_WASM: Stacktrace: \n");
-								print (err.stack);
-
-								var wasm_exit = Module.cwrap ('mono_wasm_exit', null, ['number']);
-								wasm_exit (1);
+				};
+
+				attemptNextSource = function () {
+					if (sourceIndex >= sourcesList.length) {
+						var msg = "MONO_WASM: Failed to load " + asset.name;
+						try {
+							var isOk = asset.is_optional ||
+								(asset.name.match (/\.pdb$/) && MONO.mono_wasm_ignore_pdb_load_errors);
+
+							if (isOk)
+								console.log (msg);
+							else {
+								console.error (msg);
+								throw new Error (msg);
 							}
+						} finally {
+							onPendingRequestComplete ();
+						}
+					}
+
+					var sourcePrefix = sourcesList[sourceIndex];
+					sourceIndex++;
+
+					// HACK: Special-case because MSBuild doesn't allow "" as an attribute
+					if (sourcePrefix === "./")
+						sourcePrefix = "";
+
+					var attemptUrl;
+					if (sourcePrefix.trim() === "") {
+						if (asset.behavior === "assembly")
+							attemptUrl = locateFile (args.assembly_root + "/" + asset.name);
+						else
+							attemptUrl = asset.name;
+					} else {
+						attemptUrl = sourcePrefix + asset.name;
+					}
+
+					try {
+						if (asset.name === attemptUrl) {
+							if (ctx.tracing)
+								console.log ("Attempting to fetch '" + attemptUrl + "'");
 						} else {
-							load_runtime (vfs_prefix, enable_debugging);
+							if (ctx.tracing)
+								console.log ("Attempting to fetch '" + attemptUrl + "' for", asset.name);
 						}
-						MONO.mono_wasm_runtime_ready ();
-						loaded_cb ();
+						var fetch_promise = fetch_file_cb (attemptUrl);
+						fetch_promise.then (handleFetchResponse);
+					} catch (exc) {
+						console.error ("MONO_WASM: Error fetching " + attemptUrl, exc);
+						attemptNextSource ();
 					}
-				});
+				};
+
+				attemptNextSource ();
 			});
 		},
 
+		// Performs setup for globalization.
+		// @globalization_mode is one of "icu", "invariant", or "auto".
+		// "auto" will use "icu" if any ICU data archives have been loaded,
+		//  otherwise "invariant".
+		mono_wasm_globalization_init: function (globalization_mode) {
+			var invariantMode = false;
+
+			if (globalization_mode === "invariant")
```

Comment:
How will I enable invariant mode in a blazor app?

Reply:
that's a question for the blazor team. the default mode is 'auto', where it will enable invariant unless icu data has been loaded. you can explicitly select icu-only or invariant-only via this globalization_init API

Reply:
The common way to set invariant mode for Blazer will be with using existing msbuild property called `InvariantGlobalization` to true

Reply:
Do the app needs to be published trimmed for that to work?